plugins/admin/pkg/command/registry/registry.go

```diff
@@ -47,15 +47,26 @@ type registryCred struct {
 // Auths is a map of docker credentials indexed by server url
 type Auths map[string]registryCred
 
+type registryAdminParams struct {
+	AdminParams    *pkg.AdminParams
+	Namespace      string
+	ServiceAccount string
+}
+
 // NewPrivateRegistryCmd represents the privateRegistry command
 func NewPrivateRegistryCmd(p *pkg.AdminParams) *cobra.Command {
+	registryAdminParams := &registryAdminParams{
+		AdminParams: p,
+	}
 	var privateRegistryCmd = &cobra.Command{
 		Use:   "registry",
 		Short: "Manage registry",
 		Long:  `Manage registry used by Knative service deployment`,
 	}
-	privateRegistryCmd.AddCommand(NewRegistryAddCommand(p))
-	privateRegistryCmd.AddCommand(NewRegistryRmCommand(p))
+	privateRegistryCmd.PersistentFlags().StringVarP(&registryAdminParams.Namespace, "namespace", "n", "", "the namespace to manage registries")
```

Comment:
kn already has a utility method `AddNamespaceFlags` to manage the namespace flag. The utility will add both `namespace` and `all-namespaces` to commands. It is used by all sub commands of `kn`. Since it is a kn plugin, I will suggest to follow the same way of all other kn commands. 
Refer to https://github.com/knative/client/blob/master/pkg/kn/commands/namespaced.go#L26 and see how other commands are using this method. Here is an example: https://github.com/knative/client/blob/master/pkg/kn/commands/service/create.go#L136

Reply:
I'm confused with the description "the namespace to manage registries".
Or the namespace here just means the one ServiceAccount locates in?

Reply:
@daisy-ycguo Thanks for point this out. I was originally thinking that both `add` and `remove` commands has this namespace flag, then maybe I should make the flags that `add` and `remove` share into persistent flags that belongs to the upper comamnd `registry`. But after reading other kn commands, I agree that it is better to follow theirs.

Reply:
@zhanggbj Changed to use the kn client `AddNamespaceFlags` func associate with its description("Specify the namespace to operate in").